docs/computing/containers/overview.md

```diff
@@ -1,71 +1,251 @@
-network, etc. One could say that all processes, except the kernel, live inside a bubble
-(i.e, host environment), created by the kernel. Normally, when you start new processes,
-the kernel copies the environment from the parent process to the new child processes.
-The child processes live inside the same bubble. However, when starting a new process,
-you can ask the kernel to show a completely different environment to the child process.
-This is called "running the application in a container". In particular, you can ask the
-kernel to show a completely different root file system to the new process. The file
-system can come from a single file, "a container image", that the kernel just shows
-to the new process as a regular file system. This file system image can contain
-completely different files from the file system of the parent process. Effectively,
-this allows packaging all the necessary components of a given Linux distribution and
-all the software dependencies for an application into a single file, which can be run
-in any Linux machine with a suitable kernel and virtualization support. In cloud
-environments, this makes it possible to run applications transparently irrespective
-of the Linux distribution. In HPC environments this can also mitigate the above outlined
-parallel file system performance issues.
-
-## More information on creating and running containers at CSC
-
-- [Creating HPC-friendly containers](creating.md)
-- [Running existing containers](run-existing.md)
-- [Easy-to-use container wrapper tool](tykky.md)
+---
+description: Instructions for building and running Apptainer containers in CSC supercomputers.
+---
+
+# Apptainer containers
+
+In this section, we provide instructions on how to build and run containers using [Apptainer](https://apptainer.org/) with fakeroot enabled in HPC clusters without unprivileged user namespaces.
+We explain the special aspects of building and running containers on Puhti and Mahti clusters including how to set up the build environment, how to invoke the build commands, how to write container definition files and how to run containers.
+For general instructions about building and running containers, we recommend that users read the official [Apptainer documentation](https://apptainer.org/docs/user/main/index.html).
+
+Some reasons for using containers on HPC clusters are:
+
+- Containers improve startup times and avoid I/O bottlenecks with the parallel file system, that is Lustre in Puhti and Mahti, for applications that consist of a large number of files or load many shared libraries on startup such as Python, R, and MATLAB.
+  This is because Apptainer uses the Singularity Image Format (SIF) which packs the container files into a single SquashFS file.
+
+- Running containerized software is reproducible because the container image is immutable.
+
+- Container build definitions capture more comprehensively what is installed into the container and how it is installed.
```

Comment:
Again unsure if we want to mention it, but may-bei. A big reason in GIS to use containers is the ability to use different Linux variant and therefore Ubuntu packages.